go.sh

```diff
@@ -2,15 +2,22 @@
 
 set -e
 
+# This follows a different path to the other tests/shell, since the e2e script
+# spins up its own containers as needed.
+if [ $1 = "e2e" ]
+then
+  ./bin/e2e.sh
+  exit
+fi
```

Comment:
Your comment above is correct that this is quite a different path/use case than the other things that we use `go.sh` for. With that in mind, I think it might better for it to just stay as a separate thing that can be run on its own, rather than making it a 'special case' of the go script.

Reply:
Keen to revert this change, then.